File: practice/algorithms/implementation/minimum_distances/MinimumDistances.scala
object MinimumDistances extends App {
  val lines = Source.stdin.getLines().toList
  val array = lines(1).split(" ").map(_.toInt)

  val indexPairs = array.indices.flatMap(i => array.indices.slice(i + 1, array.length).map(j => (i, j)))
  val equalValues = indexPairs.filter(pair => array(pair._1) == array(pair._2))
  val distances = equalValues.map(pair => (pair._1 - pair._2).abs)

  val result = if(distances.isEmpty) -1 else distances.min
  println(result)
}

File: practice/algorithms/implementation/minimum_distances/minimum_distances.rb
n = gets.to_i
arr = gets.split(" ").map {|i| i.to_i}
min = 99999999
for i in 0..(n-1)
  for j in (i+1)..n
    min = [min, (j - i).abs].min if (arr[j] == arr[i])
  end
end

min = -1 if(min == 99999999)
puts min